src/openrct2/world/Climate.cpp

```diff
@@ -241,20 +256,20 @@ static void climate_determine_future_weather(sint32 randomDistribution)
     // Generate a random variable with values 0 up to DistributionSize-1 and chose weather from the distribution table accordingly
     const WeatherTransition * transition = &ClimateTransitions[gClimate][month];
     sint8 nextWeather = transition->Distribution[((randomDistribution & 0xFF) * transition->DistributionSize) >> 8];
-    gClimateNextWeather = nextWeather;
+    gClimateNext.Weather = nextWeather;
 
     const auto nextWeatherState = &ClimateWeatherData[nextWeather];
-    gClimateNextTemperature = transition->BaseTemperature + nextWeatherState->TemperatureDelta;
-    gClimateNextWeatherEffect = nextWeatherState->EffectLevel;
-    gClimateNextWeatherGloom = nextWeatherState->GloomLevel;
-    gClimateNextRainLevel = nextWeatherState->RainLevel;
+    gClimateNext.Temperature = transition->BaseTemperature + nextWeatherState->TemperatureDelta;
+    gClimateNext.WeatherEffect = nextWeatherState->EffectLevel;
+    gClimateNext.WeatherGloom = nextWeatherState->GloomLevel;
+    gClimateNext.RainLevel = nextWeatherState->RainLevel;
 
     gClimateUpdateTimer = 1920;
 }
 
 static void climate_update_rain_sound()
 {
-    if (gClimateCurrentWeatherEffect == 1 || gClimateCurrentWeatherEffect == 2)
+    if (gClimateCurrent.WeatherEffect == 1 || gClimateCurrent.WeatherEffect == 2)
```

Comment:
Perhaps introducing an enum for `WeatherEffect` would be a good idea?

Reply:
done